test/general/test_entrances.py

```diff
@@ -15,25 +15,27 @@ def get_entrance_name_to_source_and_target_dict(world: World):
         gen_steps = ("generate_early", "create_regions", "create_items", "set_rules", "connect_entrances")
         additional_steps = ("generate_basic", "pre_fill")
 
-        for game_name, world_type in AutoWorldRegister.world_types.items():
-            with self.subTest("Game", game_name=game_name):
-                multiworld = setup_solo_multiworld(world_type, gen_steps)
+        for game_name, testable_world in AutoWorldRegister.testable_worlds.items():
```

Comment:
Given how often this pattern is used, it seems like it would be desirable to make it easier to reuse this as it's getting very bulky and inconvenient to use. It'll also make future refactoring easier.

One option that comes to mind but may be neither optimal nor trivial is to make an decorator that automatically implements this pattern, allowing each test to be implemented as a function of world/options. The iteration and subtests would then be handled opaquely under the hood for a majority of tests